app/code/Vladimirl/Chatter/Controller/Submit/Submit.php

```diff
@@ -57,17 +98,56 @@ public function generateHash(): string
     public function execute()
     {
         try {
-            $request = $this->getRequest();
-            $websiteId = (int)$this->storeManager->getWebsite()->getId();
-            $authorType = 'customer';
-            $authorName = 'anonymous';
+            if (!$this->formKeyValidator->validate($this->getRequest())) {
+                throw new LocalizedException(__('Something went wrong!'));
+            }
+
+            $customerId = (int) $this->customerSession->getId();
+            $websiteId = (int) $this->storeManager->getWebsite()->getId();
+
+            if (!$chatHash = $this->customerSession->getChatHash()) {
+                $this->customerSession->setChatHash($this->generateHash());
+            }
+
+            if ($customerId) {
+                $transaction = $this->transactionFactory->create();
+                $authorType = 'customer';
```

Comment:
@Vladimir-L Should move `$authorType` values to the constants in the `ChatMessage` model